demo/tlschannel.js

```diff
@@ -0,0 +1,101 @@
+
+
+// A wrapper that combines a WebSocket to the channelserver
+// with a TLS Connection for "encryption".
+//
+// This is mainly to help us test out the API and data flow
+// on something approaching real production usage patterns.
+
+const CHANNEL_SERVER = 'wss://dev.channelserver.nonprod.cloudops.mozgcp.net/v1/ws/'
+
+const FAKE_PSK = tls.utf8ToBytes('fake psk');
+const FAKE_PSK_ID = tls.utf8ToBytes('fake psk id');
+
+class InsecureTLSChannel {
+
+    constructor() {
+        this.socket = null
+        this.tlsconn = null
+        this.channelId = null
+        this.onReceive = () => { }
+    }
+
+    static create() {
+        const self = new this()
+        self._makeSocket(CHANNEL_SERVER)
+        return new Promise((resolve, reject) => {
+            const onChannelCreation = async event => {
+                try {
+                    // Receive the channelID for the newly-created channel.
+                    self.socket.removeEventListener('message', onChannelCreation)
+                    const { channelid } = JSON.parse(event.data)
```

Comment:
channelId ?(casing)

Reply:
do we want to throw on json parse fail?

Reply:
yes we do; I *think* this should throw automatically and get routed to `reject`...